homeassistant/components/tts/__init__.py

```diff
@@ -117,19 +117,14 @@ class TTSCache(TypedDict):
 
 
 @callback
-def async_resolve_engine(hass: HomeAssistant, engine: str | None) -> str | None:
-    """Resolve engine.
+def async_default_engine(hass: HomeAssistant) -> str | None:
```

Comment:
For stt we preferred the legacy providers over entities. Shouldn't we prefer entities over legacy providers like we do here?